### PR TITLE
feat(ceph_exporter): Add CI to build images on PRs, branches, pushes, and tags, as well as release.

### PR DESCRIPTION
**Pull request by ADustyOldMuffin**
This should release new tags on merge to main, and this will also have this repo building on PRs, Branches, Tags, and pushes. The tags will be branch names, the tag value, or latest (tip of main).

### CODE REVIEW

File: .github/workflows/license.yml [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-          go-version: '1.20'
-
-      - name: Download addlicense
-        run: go get -d github.com/google/addlicense
-
-      - name: Build addlicense
-        run: go build -o addlicense github.com/google/addlicense
-
-      - name: Check file licenses
-        run: ./addlicense -c DigitalOcean -l apache -y $(date +"%Y") --check .
```
Comment (robbat2) [inline .github/workflows/license.yml]: what's the replacement for this code?

File: .github/workflows/run_build.yml [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
-    - uses: actions/setup-go@v2
-      with:
-        go-version: '1.20'
-
-    - name: Download Ceph repository key
-      run: wget -q 'https://download.ceph.com/keys/release.asc'
-    - name: Install Ceph repository key
-      run: sudo apt-key add release.asc
-    - name: Declare Ceph Nautilus repository
-      run: echo deb https://download.ceph.com/debian-nautilus/ $(lsb_release -sc) main | sudo tee /etc/apt/sources.list.d/ceph.list
```
Comment (robbat2) [inline .github/workflows/run_build.yml:40]: wow, this really still tied to N release?

**Review by robbat2 — APPROVED** [review]

File: .github/workflows/run_build.yml (hunk 2 above)
Comment (ADustyOldMuffin) [reply .github/workflows/run_build.yml:40]: They add naut specifically to just enable support, I don't think it's tied to the release exactly.

File: .github/workflows/license.yml (hunk 1 above)
Comment (ADustyOldMuffin) [reply .github/workflows/license.yml]: I accidentally removed this, I can put it back.

**Pull request merged**